<a href="https://colab.research.google.com/github/allyracho/IST-Group-Assignments/blob/master/Group_Project_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Project Name: NFL DATABSE**

Created By: 

*   Jiahua Li
*   Yang Fei
*   Ally Racho
*   Vinh Nguyen





**Project Description:**

The group created a database for the National Football League, NFL. We as a team are fascinated with the NFL and felt that no database includes all the information we would like to analyze. This idea is unique because it will be a combination of data on player and team performance as well as the revenue intake. I will be looking at it from the perspective of a team owner and when showing statistics of players, it would be similar to showing how well an employee is performing or how marketable an item would be. A typical business day for the analyst would include taking statistics on previously played games, updating the database, and reporting any changes made. Through discussions and brainstorming, the group settled on five topics to include. The five are team, headcoach, highest-paid players, MVP winners, and quarterback rating. The team included a revenue table in one ER diagram but later removed it and added ticket and merchandise sales to the team table as we felt it consolidated the information. Our group researched the statistics to be included and ensured accuracy amongst all tables. Our ER diagram had numerous revisions included adding and removing tables and fixing relationships. After these changes, the group settled on our final ER diagram pictured below. 

**Video**

In [0]:
from IPython.display import IFrame
IFrame("https://www.youtube.com/embed/vqjSqhxqWHc", 560, 315)

**ER Diagram**

![alt text](https://raw.githubusercontent.com/allyracho/IST-Group-Assignments/master/G201.jpg)


**Description of ER Diagram**


Each table is set up for a different entity. The first table will show the highest paid player of each team. This will give insight into how long the highest paid players for each team will be around. The attributes of this entity are the player's name, team, date of birth, position, and net worth. A constraint will be the range of ages. The second table will be the QBR, total quarterback rating, for the starting quarterbacks on each team. The QBR is a measure out of 158.3 and takes into account metrics on passing plays than passer rating such as strength of opposition and the clutch factor. This will show how well the quarterbacks for each team is performing and how important they are for the team. The attributes will be player's name, team, completion percentage, and year. Within these attributes, there is constraints  on the years active (1990-2019). Our third table is dedicated towards the head coach of each team and how long they were hired for the team. Our group felt this was a good entity as it indicates how well the team has performed in the past and whether or not they kept the head coach. Attributes for the table include the name of coach, team coached, years present, and the number of wins. Constraints will be for the amount of years (1920-2019). The fourth table will be the name and age of the last ten most valuable player winners. This statistic will show whether the younger players are starting to overtake the older players as the best in the league. It will also show that the NFL favors offensive players more than defensive players. Name, age, position, and team are the attributes for this table. A constraint such as keeping the range of ages between (18-50) and only checking between the years (2009-2019). Our final table is dedicated towards the team with team name being the primary key. Our entities are: number of players, coach, location, number of wins, ticket sales, and merchandise sales. Originally the diagram included a revenue table, but after consideration, it was merged with the team table. The relationships are pictured above. Using these entities, attributes, and constraints our goals can hopefully be achieved and create a successful database for the user.

**How Database was Created:**

First of all, we knew the kinds of data we are interested in and we need to find out, and we used the resources from the internet for finding the data. After that, we collected those data and information for creating tables. For making it be more clear, we classified those data into different categories and created tables for each of the categories, and the tables also helped us to create the database easier. The final step is to insert the data into the database by SQL in our projects and we made sure it was working successfully, and end-users can see the results they want to search on their own.

**Create Database and Tables**

In [0]:
import sqlite3 
connectionToDatabase = sqlite3.connect('NFL') 

cursor = connectionToDatabase.cursor()
cursor.execute('''drop table IF EXISTS HeadCoach''')
connectionToDatabase.commit()

cursor.execute('''create table HeadCoach (coachID text PRIMARY KEY, F_Name text, L_Name text, TeamName text, YearsPresent int, NumberofWins int)''')
connectionToDatabase.commit()

cursor = connectionToDatabase.cursor()
cursor.execute('''drop table IF EXISTS HighestPaidPlayer''')
connectionToDatabase.commit()

cursor.execute('''create table HighestPaidPlayer (playerID text PRIMARY KEY,  F_Name text, L_Name text, TeamName text, position text, NetWorth int, DOB text)''')
connectionToDatabase.commit()

cursor.execute('''drop table IF EXISTS Team''')
connectionToDatabase.commit()

cursor.execute('''create table Team (TeamName text PRIMARY KEY, Players int, Coach text, Location text, NumberofWins int, TicketSales real, MerchaindiseSales real)''')
connectionToDatabase.commit()

cursor = connectionToDatabase.cursor()
cursor.execute('''drop table IF EXISTS MVPWinners''')
connectionToDatabase.commit()

cursor.execute('''create table MVPWinners (PlayerID text PRIMARY KEY, F_Name text, L_Name text, TeamName text, Position text, AgeofWinning int)''')
connectionToDatabase.commit()

cursor = connectionToDatabase.cursor()
cursor.execute('''drop table IF EXISTS QBR''')
connectionToDatabase.commit()

cursor.execute('''create table QBR (PlayerID text PRIMARY KEY, F_Name text, L_Name text, TeamName text, Completionpercentage real, YearsActive real)''')
connectionToDatabase.commit()

**Insert Data into Tables**

In [0]:
cursor.execute('''insert into HeadCoach values ('46', 'Doug', 'Pederson', 'Philidelphia Eagles', 4, 37);''')
connectionToDatabase.commit()
cursor.execute('''insert into HeadCoach values ('30', 'Bill', 'Belichick', 'New England Patriots', 19, 202);''')
connectionToDatabase.commit()
cursor.execute('''insert into HeadCoach values ('12', 'Mike', 'Tomlin', 'Pittsburg Steelers', 12, 129);''')
connectionToDatabase.commit()
cursor.execute('''insert into HeadCoach values ('23', 'Sean', 'McVay', 'Los Angeles Rams', 2, 29);''')
connectionToDatabase.commit()
cursor.execute('''insert into HeadCoach values ('27', 'Adam', 'Gase', 'New York Jets', 1, 4);''')
connectionToDatabase.commit()

In [0]:
cursor.execute('''insert into HighestPaidPlayer values ('9', 'Eli', 'Manning', 'New York Giants', 'QB', 252300000, '10/08/76');''')
connectionToDatabase.commit()
cursor.execute('''insert into HighestPaidPlayer values ('32', 'Peyton', 'Manning', 'Indianapolis Colts', 'QB', 248700000, '03/22/78');''')
connectionToDatabase.commit()
cursor.execute('''insert into HighestPaidPlayer values ('99', 'Drew', 'Brees', 'New Orleans Saints', 'QB', 244700000, '05/13/77');''')
connectionToDatabase.commit()
cursor.execute('''insert into HighestPaidPlayer values ('65', 'Tom', 'Brady', 'New England Patriots', 'QB', 235200000, '11/10/80');''')
connectionToDatabase.commit()
cursor.execute('''insert into HighestPaidPlayer values ('14', 'Ben', 'Roethlisberger', 'Pittsburgh Steelers', 'QB', 232300000, '06/30/76');''')
connectionToDatabase.commit()

In [0]:
insertStatement =  """insert into Team values ('Philidelphia Eagles', '12', '46', 'Philidephia, PA', 37, 78000000.00, 198000000.00)""" 
cursor.execute(insertStatement)
insertStatement =  """insert into Team values ('New England Patriots', '23', '30', 'Boston, MA', 202, 104000000.00, 360000000.00)""" 
cursor.execute(insertStatement)
insertStatement =  """insert into Team values ('Pittsburg Steelers', '34', '12', 'Pittsburgh, PA', 129, 85000000.00, 154000000.00)""" 
cursor.execute(insertStatement)
insertStatement =  """insert into Team values ('Los Angeles Rams', '45', '23', 'Los Angeles, CA', 29, 70000000.00, 93000000.00)""" 
cursor.execute(insertStatement)
insertStatement = """insert  into Team values ('New York Jets', '46','27','New York, NY', 4, 65000000.00, 81300000.00)"""
connectionToDatabase.commit()

In [0]:
cursor.execute('''insert into MVPWinners values ('20', 'Patrick', 'Mahomes', 'Kansas City Chiefs', 'QB', 24);''')
connectionToDatabase.commit()
cursor.execute('''insert into MVPWinners values ('3', 'Tom', 'Brady', 'New England Patriots', 'QB', 41);''')
connectionToDatabase.commit()
cursor.execute('''insert into MVPWinners values ('122', 'Matt', 'Ryan', 'Atlanta Falcons', 'QB', 32);''')
connectionToDatabase.commit()
cursor.execute('''insert into MVPWinners values ('75', 'Cam', 'Newton', 'Carolina Panthers', 'QB', 27);''')
connectionToDatabase.commit()
cursor.execute('''insert into MVPWinners values ('12', 'Aaron', 'Rodgers', 'Green Bay Packers', 'QB', 31);''')
connectionToDatabase.commit()

In [0]:
cursor.execute('''insert into QBR values ('12', 'Carson', 'Werkz', 'Philadelphia Eagles', '65.5%', 5);''')
connectionToDatabase.commit()
cursor.execute('''insert into QBR values ('23', 'Mason', 'Pudolph', 'Pittsburgh Steelers', '78.7%', 3);''')
connectionToDatabase.commit()
cursor.execute('''insert into QBR values ('34', 'Kirk', 'Cousins', 'Minnesota Vikings', '35.5', 4);''')
connectionToDatabase.commit()
cursor.execute('''insert into QBR values ('45', 'Josh', 'Rosen', 'Miami Dophins', '39.4', 3);''')
connectionToDatabase.commit()
cursor.execute('''insert into QBR values ('20', 'Luke', 'Faulk', 'New York Jets', '26.2', 2);''')
connectionToDatabase.commit()

**Checking My Tables:** 

In [0]:
cursor.execute('''select * from Team''')
for row in cursor.fetchall():
  print (row)

('Philidelphia Eagles', 12, '46', 'Philidephia, PA', 37, 78000000.0, 198000000.0)
('New England Patriots', 23, '30', 'Boston, MA', 202, 104000000.0, 360000000.0)
('Pittsburg Steelers', 34, '12', 'Pittsburgh, PA', 129, 85000000.0, 154000000.0)
('Los Angeles Rams', 45, '23', 'Los Angeles, CA', 29, 70000000.0, 93000000.0)


In [0]:
cursor.execute('''select * from QBR''')
for row in cursor.fetchall():
  print (row)

('12', 'Carson', 'Werkz', 'Philadelphia Eagles', '65.5%', 5.0)
('23', 'Mason', 'Pudolph', 'Pittsburgh Steelers', '78.7%', 3.0)
('34', 'Kirk', 'Cousins', 'Minnesota Vikings', 35.5, 4.0)
('45', 'Josh', 'Rosen', 'Miami Dophins', 39.4, 3.0)
('20', 'Luke', 'Faulk', 'New York Jets', 26.2, 2.0)


In [0]:
cursor.execute('''select * from HeadCoach''')
for row in cursor.fetchall():
  print (row)

('46', 'Doug', 'Pederson', 'Philidelphia Eagles', 4, 37)
('30', 'Bill', 'Belichick', 'New England Patriots', 19, 202)
('12', 'Mike', 'Tomlin', 'Pittsburg Steelers', 12, 129)
('23', 'Sean', 'McVay', 'Los Angeles Rams', 2, 29)
('27', 'Adam', 'Gase', 'New York Jets', 1, 4)


**Scenario for an Average User**

Jim Smith is a die-hard Philadelphia Eagles fan and also a business major at Penn State. He is fascinated by the statistics of teams and likes to see how many wins each has. Jim checks daily how his team is doing overall in the league. With this database, everything will be available at the touch of a button all in one place. Normally Jim has to check multiple websites for his statistics, but not anymore. Lately he is interested in knowing MVP Winners as he is working on a project in school for it. Instead of having to go to different websites, our database lets the user see who the winners were and even allows them to filter the results by year and age. With this database, Jim has all the information he ever needed about the NFL all in one place and does not need the hassle of having to search for it anymore.

**Queries**

What is the total number of wins among the 5 teams?


In [0]:
cursor.execute('''select SUM(NumberofWins) from Team''')
for row in cursor.fetchall():
  print (row)

(397,)


The group decided on this question because Jim enjoys looking at the statistics among teams. The total among just five teams in our database is 397. This number will continue to increase overtime as more teams are added to the database and teams are created. Jim used to have to go to multiple websites to find out this answer, but not anymore.

How many of the MVP Winners are above the age of 25?

In [0]:
cursor.execute('''select Count(AgeofWinning) from MVPWinners WHERE AgeofWinning> 25 ''')
for row in cursor.fetchall():
  print (row)

(4,)


This question is important because the team and Jim are interested in seeing how many players are keeping good form with their growing age. The group is also interesting in seeing if more older players are winning. The results match up to our expectations. We can see that there are a lot of players winning MVP older than 25.

Where is the location of each team?

In [0]:
cursor.execute('''select h.TeamName, t.Location from Team t, HeadCoach h where t.Coach=h.coachID''')
for row in cursor.fetchall():
  print (row)
  

('Philidelphia Eagles', 'Philidephia, PA')
('New England Patriots', 'Boston, MA')
('Pittsburg Steelers', 'Pittsburgh, PA')
('Los Angeles Rams', 'Los Angeles, CA')


The group felt this to be a suitable question as it easily locates the team and displays which city and state it is in. Our expections for what would be shown were correct, with the team name being first, and the location following after. This question allows for any team to be found conveniently if their team name does not necessarily match their location.